packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -26,14 +26,44 @@ typedef PointerExitEventListener = void Function(PointerExitEvent event);
 /// Used by [MouseTrackerAnnotation], [MouseRegion] and [RenderMouseRegion].
 typedef PointerHoverEventListener = void Function(PointerHoverEvent event);
 
+// A key, so that (_IdenticalAnnotationKey(a) == _IdenticalAnnotationKey(b)) is equivalent to
+// identical(a, b).
+class _IdenticalAnnotationKey implements Key {
+  _IdenticalAnnotationKey(this.value);
+
+  final MouseTrackerAnnotation value;
+
+  // Don't return `identityHashCode(value)` directly since the hash of this key
+  // needs to be different from that of `value`. Do a fast conversion with a
+  // random mask, since we don't need a secure hashing algorithm for a private
+  // class.
+  @override
+  int get hashCode => _hashCodeMask ^ identityHashCode(value);
+  static const int _hashCodeMask = 0xc366d8df10ee8;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    return other is _IdenticalAnnotationKey && identical(other.value, value);
```

Comment:
Please use the common boilerplates:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator-and-hashcode